crates/router/src/core/payment_methods/cards.rs

```diff
@@ -895,14 +1018,32 @@ pub async fn get_client_secret_or_add_payment_method(
 }
 
 #[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
-#[instrument(skip_all)]
-pub async fn get_client_secret_or_add_payment_method(
-    _state: &routes::SessionState,
-    _req: api::PaymentMethodCreate,
-    _merchant_account: &domain::MerchantAccount,
-    _key_store: &domain::MerchantKeyStore,
-) -> errors::RouterResponse<api::PaymentMethodResponse> {
-    todo!()
+pub async fn create_pm_additional_data_update(
```

Comment:
Payment method related function should not be in cards.rs, instead it should be in payment_method.rs file